pkg/appgw/backendhttpsettings.go

```diff
@@ -23,43 +23,67 @@ const (
 	DefaultConnDrainTimeoutInSec = 30
 )
 
-func newBackendIds(ingressList []*v1beta1.Ingress) map[backendIdentifier]interface{} {
+func newBackendIds(ingressList []*v1beta1.Ingress, serviceList []*v1.Service) map[backendIdentifier]interface{} {
 	backendIDs := make(map[backendIdentifier]interface{})
 	for _, ingress := range ingressList {
 		if ingress.Spec.Backend != nil {
-			glog.Infof("Ingress spec has no backend. Adding a default.")
-			backendIDs[generateBackendID(ingress, nil, nil, ingress.Spec.Backend)] = nil
+			backendID := generateBackendID(ingress, nil, nil, ingress.Spec.Backend)
+			glog.Info("Ingress spec has no backend. Adding default backend:", backendID.serviceKey(), backendID.serviceFullName())
+			backendIDs[backendID] = nil
 		}
 		for ruleIdx := range ingress.Spec.Rules {
 			rule := &ingress.Spec.Rules[ruleIdx]
-			glog.Infof("Working on ingress rule #%d: host='%s'", ruleIdx+1, rule.Host)
+			glog.V(5).Infof("Working on ingress rule #%d: host='%s'", ruleIdx+1, rule.Host)
 			if rule.HTTP == nil {
 				// skip no http rule
-				glog.Infof("Skip rule #%d for host '%s' - it has no HTTP rules.", ruleIdx+1, rule.Host)
+				glog.V(5).Infof("Skip rule #%d for host '%s' - it has no HTTP rules.", ruleIdx+1, rule.Host)
 				continue
 			}
 			for pathIdx := range rule.HTTP.Paths {
 				path := &rule.HTTP.Paths[pathIdx]
-				glog.Infof("Working on path #%d: '%s'", pathIdx+1, path.Path)
-				backendIDs[generateBackendID(ingress, rule, path, &path.Backend)] = nil
+				glog.V(5).Infof("Working on path #%d: '%s'", pathIdx+1, path.Path)
+				backendID := generateBackendID(ingress, rule, path, &path.Backend)
+				glog.Info("Adding backend:", backendID.serviceKey(), backendID.serviceFullName())
+				backendIDs[backendID] = nil
 			}
 		}
 	}
-	return backendIDs
+
+	finalBackendIDs := make(map[backendIdentifier]interface{})
+	serviceSet := newServiceSet(&serviceList)
+	// Filter out backends, where Ingresses reference non-existent Services
+	for be := range backendIDs {
+		if _, exists := serviceSet[be.serviceKey()]; !exists {
+			glog.Errorf("Ingress %s/%s references non existent Service %s. Please correct the Service section of your Kubernetes YAML", be.Ingress.Namespace, be.Ingress.Name, be.serviceKey())
```

Comment:
Logically, validation that a service exists or not should not happen as a part of `newBackendIds`.

Reply:
That's an excellent point.
Let's move this chunk into https://github.com/Azure/application-gateway-kubernetes-ingress/pull/201 and open https://github.com/Azure/application-gateway-kubernetes-ingress/pull/201 for review